File: config_yaml/ecs_stack.py
from aws_cdk import Stack
from aws_cdk.aws_ec2 import Vpc, InstanceType
from aws_cdk.aws_ecs import Cluster, Ec2Service, Ec2TaskDefinition, ContainerImage
from constructs import Construct

class EcsServiceStack(Stack):
    def __init__(self, scope: Construct, construct_id: str, config: dict, **kwargs):
        super().__init__(scope, construct_id, **kwargs)

        vpc_cfg = config["vpc"]
        ecs_cfg = config["ecs"]

        vpc = Vpc(self, "Vpc", max_azs=vpc_cfg["maxAzs"])

        cluster = Cluster(
            self,
            "Cluster",
            vpc=vpc,
            cluster_name=ecs_cfg["clusterName"]
        )

        cluster.add_capacity(
            "DefaultAutoScalingGroup",
            instance_type=InstanceType(ecs_cfg["instanceType"])
        )

        task_def = Ec2TaskDefinition(self, "TaskDef")
        task_def.add_container(
            "AppContainer",
            image=ContainerImage.from_registry(ecs_cfg["dockerImage"]),
            memory_limit_mib=256
        )

        Ec2Service(
            self,
            "EcsService",
            cluster=cluster,
            task_definition=task_def,
            service_name=ecs_cfg["serviceName"]
        )

File: cdk-cross-account-ecs-java-springboot/lib/ecs-stack.ts
import * as cdk from 'aws-cdk-lib';
import * as ecs from 'aws-cdk-lib/aws-ecs';
import * as ec2 from 'aws-cdk-lib/aws-ec2';
import { Construct } from 'constructs';

export class EcsStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const vpc = new ec2.Vpc(this, 'Vpc', { maxAzs: 2 });

    const cluster = new ecs.Cluster(this, 'Cluster', { vpc });

    const taskDef = new ecs.FargateTaskDefinition(this, 'TaskDef', {
      cpu: 512,
      memoryLimitMiB: 1024,
    });

    taskDef.addContainer('JavaAppContainer', {
      image: ecs.ContainerImage.fromRegistry(
        '111111111111.dkr.ecr.ap-southeast-2.amazonaws.com/java-app-repo:latest'
      ),
      portMappings: [{ containerPort: 8080 }],
      logging: ecs.LogDrivers.awsLogs({ streamPrefix: 'JavaApp' }),
    });

    new ecs.FargateService(this, 'FargateService', {
      cluster,
      taskDefinition: taskDef,
      desiredCount: 1,
      assignPublicIp: true,
    });
  }
}
